core/sql/executor/ExHbaseSelect.cpp

```diff
@@ -444,6 +444,7 @@ ExWorkProcRetcode ExHbaseScanSQTaskTcb::work(short &rc)
   Lng32 retcode = 0;
   rc = 0;
   Lng32 remainingInBatch = batchSize_;
+  NABoolean isFirstBatch = false;
```

Comment:
OK so here this boolean is not needed for the full length of the operator. Only for the first 8K rows.
I can change the logic and maintain a row counter on the tcb. That will have impact memory write for the full length of the operator. By doing my initial trick, I was hopping that I could leverage the existing remainingInBatch counter and not create another one. But I guess if I cannot trust that the parent queue can fill up with less than an HBase_Buffer_size, I cannot use the optimization.